src/unix/getaddrinfo.c

```diff
@@ -203,12 +244,30 @@ int uv_getaddrinfo(uv_loop_t* loop,
     req->hostname = memcpy(buf + len, hostname, hostname_len);
 
   if (cb) {
+#if defined(_USE_LIBINFO)
+    int32_t get_addrinfo_result;
+    mach_port_t port;
+    get_addrinfo_result = getaddrinfo_async_start(&port,
+                                                  req->hostname,
+                                                  req->service,
+                                                  req->hints,
+                                                  uv__getaddrinfo_async_done,
+                                                  req);
+    if (get_addrinfo_result == -1) {
+      return get_addrinfo_result;
+    }
+    uv__io_init(&req->async_addrinfo_io, uv__getaddrinfo_async_work, port);
+    req->async_addrinfo_io.data = req;
```

Comment:
@cabelitos do we really need this? I saw @bnoordhuis comment about `data`, but since `req->async_addrinfo_io` is contained in `req`, to obtain the `req` given `async_addrinfo_io` it's a simpler pointer arithmetic as done by `container_of()`.

Reply:
IOW: you don't do anything in here, when you need to use from `uv__getaddrinfo_async_work` you do:

```c
uv_getaddrinfo_t* req = container_of(w, uv_getaddrinfo_t, async_addrinfo_io);
```